Update COSE_Encrypt scope and future considerations (#77)
cose-wg/draft-ietf-cose-hash-envelope: draft-ietf-cose-hash-envelope.md

## COSE_Encrypt

> Only COSE_Sign/COSE_Sign1 and COSE_Mac/COSE_Mac0 are in scope for this document. COSE_Encrypt/COSE_Encrypt0 is out of the scope of this document: there is no known use case for it. It may be covered by a future extension, which would need to address whether the hash function is applied before or after encryption, and clarify privacy considerations arising from that decision.

Reviewer: ```suggestion
Only COSE_Sign/COSE_Sign1 and COSE_Mac/COSE_Mac0 are in scope for this document. COSE_Encrypt/COSE_Encrypt0 is out of the scope of this document.
At the time of publishing, there is no known use case for COSE_Encrypt/COSE_Encrypt0. 
It may be covered by a future extension, which would address whether the hash function is applied before or after encryption, and clarify privacy considerations.
```

> Only COSE_Sign/COSE_Sign1 and COSE_Mac/COSE_Mac0 are in scope for this document. COSE_Encrypt/COSE_Encrypt0 is out of the scope of this document: there is no known use case for it. It may be covered by a future extension, which would need to address whether the hash function is applied before or after encryption, and clarify privacy considerations arising from that decision.

Author: @OR13 feedback from Henk yesterday was similar, he wanted to have a discussion with the reviewer first. I had left the PR open as a option.
